nixos/modules/services/networking/netboot.nix

```diff
@@ -0,0 +1,29 @@
+{ config, pkgs, lib }:
+
+let
+  configuration = with lib; {
+    imports = [
+        (pkgs.path + "/nixos/modules/installer/netboot/netboot-minimal.nix")
+    ];
+    ## Some useful options for setting up a new system
+    services.mingetty.autologinUser = mkForce "root";
+    # Enable sshd which gets disabled by netboot-minimal.nix
+    systemd.services.sshd.wantedBy = mkOverride 0 [ "multi-user.target" ];
+    # users.users.root.openssh.authorizedKeys.keys = [ ... ];
+    # i18n.consoleKeyMap = "de";
+  };
+
+  nixos = import (pkgs.path + "/nixos") {
+    inherit configuration;
+    # system = ...;
+  };
+in
+  pkgs.symlinkJoin {
```

Comment:
I don't think we need the symlinkJoin here:
This is what I am using:

```
  services.pixiecore = let
    nixos = import <nixpkgs/nixos> {
      configuration = { config, pkgs, lib, ... }: with lib; {
        imports = [
          <nixpkgs/nixos/modules/installer/netboot/netboot-minimal.nix>
        ];
        # Some useful options for setting up a new system
        services.mingetty.autologinUser = mkForce "root";
        # Enable sshd which gets disabled by netboot-minimal.nix
        systemd.services.sshd.wantedBy = mkOverride 0 [ "multi-user.target" ];
        # users.users.root.openssh.authorizedKeys.keys = [ ... ];
        # i18n.consoleKeyMap = "de";
      };
    };
    build = nixos.config.system.build;
  in {
    enable = true;
    openFirewall = true;
    mode = "boot";
    kernel = "${build.kernel}/bzImage";
    initrd = "${toString build.netbootRamdisk}/initrd";
    cmdLine = "init=${build.netbootIpxeScript} ${lib.concatStringsSep " " nixos.config.boot.kernelParams} debug";
    dhcpNoBind = true;
  };
```

Reply:
Also people want likely pass there own NixOS configuration to the netboot instance as well. You can have a look at `containers.<name>.config` in `nixpkgs/nixos/modules/virtualisation/containers.nix`